### PR TITLE
API-7228 Add escalatedTo optional string to Submission.Granted

### PR DESCRIPTION
**Pull request by petekirby-ee**

### CODE REVIEW

File: app/uk/gov/hmrc/apigatekeeperapprovalsfrontend/controllers/TermsOfUseFailedJourneyController.scala [hunk 1]
```diff
@@ -170,7 +200,89 @@ class TermsOfUseFailedJourneyController @Inject() (
     } yield ok
   }
 
+  def failOverridePage(applicationId: ApplicationId) = loggedInThruStrideWithApplicationAndSubmission(applicationId) { implicit request =>
+    successful(Ok(termsOfUseFailOverridePage(ViewModel(applicationId, request.application.name))))
+  }
+
+  def failOverrideAction(applicationId: ApplicationId) = loggedInThruStrideWithApplicationAndSubmission(applicationId) { implicit request =>
+    request.body.asFormUrlEncoded.getOrElse(Map.empty).get("override").flatMap(_.headOption) match {
+      case Some("yes") => successful(Redirect(uk.gov.hmrc.apigatekeeperapprovalsfrontend.controllers.routes.TermsOfUseFailedJourneyController.overrideApproverPage(applicationId)))
```
Comment (johnsgp) [inline app/uk/gov/hmrc/apigatekeeperapprovalsfrontend/controllers/TermsOfUseFailedJourneyController.scala:209]: Could we `import uk.gov.hmrc.apigatekeeperapprovalsfrontend.controllers.routes._` to keep this and similar lines shorter?

File: app/uk/gov/hmrc/apigatekeeperapprovalsfrontend/controllers/TermsOfUseFailedJourneyController.scala [hunk 2]
```diff
@@ -170,7 +200,89 @@ class TermsOfUseFailedJourneyController @Inject() (
...
+  def overrideApproverPage(applicationId: ApplicationId) = loggedInThruStrideWithApplicationAndSubmission(applicationId) { implicit request =>
+    successful(Ok(termsOfUseOverrideApproverPage(approverForm, ViewModel(applicationId, request.application.name))))
+  }
+
+  def overrideApproverAction(applicationId: ApplicationId) = loggedInThruStrideWithApplicationAndSubmission(applicationId) { implicit request =>
+    def handleValidForm(form: ApproverForm) = {
+      submissionReviewService.updateEscalatedTo(form.firstName + " " + form.lastName)(request.submission.id, request.submission.latestInstance.index).map {
+        case Some(value) => Redirect(uk.gov.hmrc.apigatekeeperapprovalsfrontend.controllers.routes.TermsOfUseFailedJourneyController.overrideNotesPage(applicationId))
+        case None        => {
+          logger.warn("Persisting override escalated to failed")
```
Comment (johnsgp) [inline app/uk/gov/hmrc/apigatekeeperapprovalsfrontend/controllers/TermsOfUseFailedJourneyController.scala:223]: This log feels vague - could it be altered to have more context when investigating, e.g. applicationId?  Similarly for other logs added in this PR.

File: it/uk/gov/hmrc/apigatekeeperapprovalsfrontend/connectors/SubmissionConnectorISpec.scala [hunk 3]
```diff
@@ -58,6 +58,7 @@ class SubmissionConnectorISpec extends BaseConnectorIntegrationISpec with GuiceO
     val markSubmission = markedSubmission
     val requestedBy    = "bob@blockbusters.com"
     val reason         = "reason"
+    val escalatedTo    = "Mr Edmund Blackadder"
```
Comment (johnsgp) [inline it/uk/gov/hmrc/apigatekeeperapprovalsfrontend/connectors/SubmissionConnectorISpec.scala:61]: Is this wise? Feels like a weak escalation point :smile:

File: app/uk/gov/hmrc/apigatekeeperapprovalsfrontend/controllers/TermsOfUseFailedJourneyController.scala (hunk 1 above)
Comment (petekirby-ee) [reply app/uk/gov/hmrc/apigatekeeperapprovalsfrontend/controllers/TermsOfUseFailedJourneyController.scala]: Fixed

File: app/uk/gov/hmrc/apigatekeeperapprovalsfrontend/controllers/TermsOfUseFailedJourneyController.scala (hunk 2 above)
Comment (petekirby-ee) [reply app/uk/gov/hmrc/apigatekeeperapprovalsfrontend/controllers/TermsOfUseFailedJourneyController.scala]: Fixed

File: it/uk/gov/hmrc/apigatekeeperapprovalsfrontend/connectors/SubmissionConnectorISpec.scala (hunk 3 above)
Comment (petekirby-ee) [reply it/uk/gov/hmrc/apigatekeeperapprovalsfrontend/connectors/SubmissionConnectorISpec.scala:61]: Lol

**Review by johnsgp — APPROVED** [review]

**Pull request merged**